src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_TYPES = 5;
+    private static final int MAX_SIDE = 5;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(FIGURE_TYPES);
+        ColorSupplier randomColor = new ColorSupplier();

```

Comment:
lets make this variable class level, not to create new instance of ColorSupplier every time we call getRandomFigure() method